### DISCUSSION TITLE
How to handle long blocking calls?

### DISCUSSION THREAD

**Discussion by hakoptak** [uNetworking/uWebSockets#1228] [Q&A] [answered]
Hi, I am new to uWebSockets and I have a use case where I need to make a http-request (using cURL) to another server.

If I use a blocking cURL call, should it respond within the HTTP_TIMEOUT_S limit?
Or am I adviced to use the non-blocking version?

And for my understanding: how does the Loop handle a blocking situation?

Thanks in advance!

**Comment by ibsusu** [comment, marked as answer]
You should use the non-blocking version.  There is a way to reuse the event loop that uWebSockets depends on but I don't know what way would fit your project.  There are examples on the curl website for utilizing epoll or libuv, I suggest following them then integrating them somehow.

**Reply by hakoptak** [reply]
Thanks for your answer. I'll look into it.
